uint8 r
uint8 g
uint8 b
float64 x
float64 y
float64 z
float64 distance